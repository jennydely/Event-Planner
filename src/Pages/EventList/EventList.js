import React, { useContext } from 'react'
import styled from 'styled-components/macro'
import resetIcon from '../../img/resetIcon.svg'
import { sortEvents } from './services/sortEvents'
import { filterEvents } from './services/filterEvents'
import { getFilteredEventList } from './services/getFilteredEventList'
import { getAvailableYears } from './services/getAvailableYears'
import EventItem from './components/EventItem'
import { UserContext } from "../../providers/UserProvider";
import useDeepCompareEffect from 'use-deep-compare-effect'

export default function EventList({
  eventArray,
  categoryFilter,
  eventFilter,
  chosenEventListFilter,
  eventSearch,
  multipleEventSearch,
  onTicketCheckboxClick,
  handleReloadButtonClick,
}) {
  const events = sortEvents(eventArray, eventFilter)
  const categoryfilteredEvents = filterEvents(events, categoryFilter)
  const filteredEventList = getFilteredEventList(
    {
      eventArray,
      eventFilter,
      chosenEventListFilter,
      eventSearch,
      multipleEventSearch,
      categoryfilteredEvents
    }
  )
  const availableYears = getAvailableYears(eventFilter, filteredEventList)
  const user = useContext(UserContext);
  useDeepCompareEffect(() => {
  }, [filteredEventList])


  return (
    <>
      {filteredEventList.length > 0 ? (
        availableYears.map((year) => (
          <EventContainer key={year}>
            <EventYearHeadline>Events {year}</EventYearHeadline>
            {filteredEventList.map(
              (event) =>
                event.eventStartDate.slice(0, 4) === year && (
                  user ?
                    <EventItem
                      event={event}
                      key={event.id}
                      id={event.id}
                      filteredEventList={filteredEventList}
                      onTicketCheckboxClick={onTicketCheckboxClick}
                    ></EventItem>
                    :
                    (<EventItem
                      event={event}
                      key={event.id}
                      id={event.id}
                    ></EventItem>)

                )
            )}
          </EventContainer>
        ))
      ) : (
          <>
            <p>No events</p>
            <ResetButton onClick={handleReloadButtonClick}>
              <img src={resetIcon} alt="reset" />
            </ResetButton>
          </>
        )}
    </>
  )
}

const EventContainer = styled.ul`
  margin: 0;
  padding: 0;
`
const EventYearHeadline = styled.h1`
  margin-bottom: 5px;
  padding: 0;
  text-align: left;
`
const ResetButton = styled.button`
  align-self: center;
`
