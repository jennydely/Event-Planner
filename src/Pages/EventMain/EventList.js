import React from 'react'
import styled from 'styled-components/macro'
import { sortEvents } from './services/sortEvents'
import { filterEvents } from './services/filterEvents'
import { getFilteredEvents } from './services/getFilteredEvents'
import { getAvailableYears } from './services/getAvailableYears'
import EventItem from './components/EventItem'

export default function EventList({
  eventArray,
  categoryFilter,
  eventFilter,
  eventSearch,
  multipleEventSearch,
  onHideButtonClick,
  onEditButtonClick,
  onDeleteButtonClick,
  onTicketCheckboxClick,
}) {
  const events = sortEvents(eventArray, eventFilter)
  const categoryfilteredEvents = filterEvents(events, categoryFilter)
  const filteredEvents = getFilteredEvents(
    eventArray,
    eventFilter,
    eventSearch,
    multipleEventSearch,
    categoryfilteredEvents
  )
  const availableYears = getAvailableYears(eventFilter, filteredEvents)

  return (
    <>
      {availableYears.map((year) => (
        <EventContainer key={year}>
          <EventYearHeadline>Events {year}</EventYearHeadline>
          {filteredEvents.map(
            (event) =>
              event.eventStartDate.slice(0, 4) === year && (
                <EventItem
                  event={event}
                  key={event.id}
                  id={event.id}
                  onHideButtonClick={onHideButtonClick}
                  onDeleteButtonClick={onDeleteButtonClick}
                  onEditButtonClick={onEditButtonClick}
                  onTicketCheckboxClick={onTicketCheckboxClick}
                ></EventItem>
              )
          )}
        </EventContainer>
      ))}
    </>
  )
}

const EventContainer = styled.ul`
  margin: 0;
  padding: 0;
`
const EventYearHeadline = styled.h1`
  margin-bottom: 5px;
  padding: 0;
  text-align: left;
`
