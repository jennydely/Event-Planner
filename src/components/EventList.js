import React from 'react'
import EventItem from './EventItem'
import styled from 'styled-components/macro'
import { sortEvents } from '../services/sortEvents'
import { filterEvents } from '../services/filterEvents'
import { getYearsOfEvents } from '../services/getYearsOfEvents'

export default function EventList({ eventArray, categoryFilter, eventFilter }) {
  const events = sortEvents(eventArray, eventFilter)
  const filteredEvents = filterEvents(events, categoryFilter)
  const availableYears = getYearsOfEvents(filteredEvents)

  return (
    <>
      {availableYears.map((year) => (
        <EventContainer key={year}>
          <EventYearHeadline>Events {year}</EventYearHeadline>
          {filteredEvents.map(
            (event) =>
              event.eventStartDate.slice(0, 4) === year && (
                <EventItem event={event} key={event.id}></EventItem>
              )
          )}
        </EventContainer>
      ))}
    </>
  )
}

const EventContainer = styled.ul`
  display: grid;
  grid-template-columns: auto 50px;
  margin: 10px;
  padding: 0;
`
const EventYearHeadline = styled.h1`
  margin-bottom: 5px;
  padding: 0;
  text-align: left;
`
