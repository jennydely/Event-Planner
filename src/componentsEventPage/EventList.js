import React from 'react'
import Event from './Event'
import styled from 'styled-components/macro'
//import {getEvents} from '../utils/services'

export default function EventList({eventArray, categoryFilter}) {
   
    eventArray.sort((event1, event2) => event1.eventDate > event2.eventDate)
    const currentYearString = new Date().getFullYear().toString()
    const availableYears = eventArray.reduce((years, event) => {
        const eventYear = event.eventDate.slice(0, 4)
        if (!years.includes(eventYear) && eventYear >= currentYearString) years.push(eventYear)
        return years
    }, [currentYearString])

    return (
        <>
            {availableYears.map(year => (
                <EventContainer key={year}>
                    <EventYearHeadline>Event {year}</EventYearHeadline>
                    {eventArray.filter(event => categoryFilter === 'all' || event.category === categoryFilter).map(event => event.eventDate.slice(0, 4) === year &&
                        <Event event={event} key={event.id}></Event>
                    )}
                </EventContainer>
            ))}
        </>
    )
}

const EventContainer = styled.ul`
margin: 10px;
padding:0;
`
const EventYearHeadline= styled.h2`
margin: 0;
margin-bottom: 5px;
padding:0;
text-align: left;
`
