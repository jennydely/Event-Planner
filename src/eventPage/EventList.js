import React from 'react';
import Event from './Event'
//import {getEvents} from '../utils/services'

const jsonEvents = require('../mockDB/events.json')

export default function EventList() {

    return (
        <ul>
            {jsonEvents.map(event =>
                <Event event={event} key={event.id}></Event>
            )}
        </ul>
    )
}