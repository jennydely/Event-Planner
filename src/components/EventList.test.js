import React from 'react'
import { render } from '@testing-library/react'
import EventList from './EventList'
import 'jest-styled-components'

describe('EventList', () => {
  const eventArray = [
    {
      id: '123',
      name: 'this is the eventname',
      location: 'this is the eventlocation',
      eventStartDate: '06/25/2020',
      eventEndDate: '06/28/2020',
      category: 'medieval',
    },
    {
      id: '321',
      name: 'this is the eventname',
      location: 'this is the eventlocation',
      eventStartDate: '06/25/2021',
      eventEndDate: '06/28/2021',
      category: 'medieval',
    },
  ]
  const categoryFilter = 'medieval'

  it('displays the year and an example event', () => {
    const { getByText } = render(
      <EventList eventArray={eventArray} categoryFilter={categoryFilter} />
    )
    setTimeout(() => {
      expect(getByText('2020')).toBeInTheDocument()
      expect(getByText(eventArray[0].name)).toBeInTheDocument()
      expect(getByText(eventArray[0].location)).toBeInTheDocument()
    }, 100)
  })
  it('renders correctly', () => {
    const tree = render(
      <EventList eventArray={eventArray} categoryFilter={categoryFilter} />
    )
    expect(tree).toMatchSnapshot()
  })
})
