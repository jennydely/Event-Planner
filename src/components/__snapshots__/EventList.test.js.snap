// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: grid;
  grid-template-columns: 316px 45px;
  margin: 0;
  padding: 0;
}

.c1 {
  margin-bottom: 5px;
  padding: 0;
  text-align: left;
}

<body>
    <div>
      <ul
        class="c0"
      >
        <h1
          class="c1"
        >
          Events 
          2020
        </h1>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="EventList__EventContainer-sc-1c1tgab-0 jFhGDi"
    >
      <h1
        class="EventList__EventYearHeadline-sc-1c1tgab-1 boyGhL"
      >
        Events 
        2020
      </h1>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
